### ISSUE TITLE
Spock Spring module does not work with Spring Test 6

### ISSUE THREAD

**Issue by Poundex** [spockframework/spock#1539]
### Describe the bug

When using Spring Test 6 (used by Spring 6/Boot 3) then the Spock Spring module does not detect the Spring tests. The tests are still picked up and an attempt to execute them is made, but the Spring Application Context is never started.

The issue seems to be caused by the class `org.springframework.test.util.MetaAnnotationUtils` which was deprecated in 5.3 being removed in 6. It looks like Spock used this utility class to discover the `@BootstrapWith` meta-annotation on the `@SpringBootTest` annotation. 

There are separate checks for `@ContextHierarchy` and `@ContextConfiguration`, so if a Spec is using these directly then the Application Context starts as expected.

**Workaround**
Add an empty `@ContextConfiguration` annotation to the Spec.

### To Reproduce

```
@SpringBootTest(classes = [DemoApplication, JustATestSpec])
class JustATestSpec extends Specification implements ApplicationListener<ApplicationReadyEvent> {
	
	static boolean contextSeenStarting = false
	
	@Override
	void onApplicationEvent(ApplicationReadyEvent event) {
		contextSeenStarting = true
	}

	void "Starts Application Context"() {
		expect:
		contextSeenStarting
	}
}

```

### Expected behavior

Application context starts

### Actual behavior

Application context doesn't start

### Java version

All

### Buildtool version

All

### What operating system are you using

Linux

### Dependencies

?

**Comment by winne42** [comment]
@leonard84 I assume that many Spock users are desperately waiting for your fix. Do you have any idea about when a new Spock version including your fix will be released?

**Comment by sbhuju61** [comment]
@winne42 The fix has already been released. I was able to successfully use it few weeks ago.
I used 2.4-M1-groovy-4.0. I believe other version also have the fix.

https://mvnrepository.com/artifact/org.spockframework/spock-spring/2.4-M1-groovy-4.0

**Comment by winne42** [comment]
@sbhuju61 but is the "M1" release supposed to be stable? Sounds like a "milestone" to me?

**Comment by sbhuju61** [comment]
> @sbhuju61 but is the "M1" release supposed to be stable? Sounds like a "milestone" to me?

It is a milestone and not as stable as final release. I have been using it since mid December of 2022 and have not run into any issues yet.

**Comment by nniesen** [comment]
This has been stuck in a Milestone Release for 6 months. Any chance it can be released or backport to 2.3. Currently, the only options appear to be not upgrading to Boot 3.x or having to depend on a Milestone dependency.
